modules/apps/client-extension/client-extension-web/src/main/resources/META-INF/resources/admin/edit_editor_configuration.jsp

```diff
@@ -0,0 +1,108 @@
+<%--
+/**
+ * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+--%>
+
+<%@ include file="/admin/init.jsp" %>
+
+<%
+EditClientExtensionEntryDisplayContext<EditorConfigurationCET> editClientExtensionEntryDisplayContext = (EditClientExtensionEntryDisplayContext)renderRequest.getAttribute(ClientExtensionAdminWebKeys.EDIT_CLIENT_EXTENSION_ENTRY_DISPLAY_CONTEXT);
+
+EditorConfigurationCET editorConfigurationCET = editClientExtensionEntryDisplayContext.getCET();
+%>
+
+<aui:field-wrapper cssClass="form-group">
+	<aui:input label="js-url" name="url" required="<%= true %>" type="text" value="<%= editorConfigurationCET.getURL() %>" />
+
+	<div class="form-text">
+		<liferay-ui:message key="enter-the-url-of-the-javascript-file-to-provide-editor-configuration-json" />
+	</div>
+</aui:field-wrapper>
+
+<div class="lfr-form-rows" id="<portlet:namespace />_portletNames_field">
+
+	<%
+	for (String _portletName : editClientExtensionEntryDisplayContext.getStrings(editorConfigurationCET.getPortletNames())) {
+	%>
+
+		<div class="lfr-form-row">
+			<aui:field-wrapper cssClass="form-group">
+				<aui:input ignoreRequestValue="<%= true %>" label="portlet-names" name="portletNames" type="text" value="<%= _portletName %>" />
+
+				<div class="form-text form-text-repeat">
+					<liferay-ui:message key="enter-portlet-name-to-apply-this-editor-configuration" />
+				</div>
+			</aui:field-wrapper>
+		</div>
+
+	<%
+	}
+	%>
+
+</div>
+
+<div class="lfr-form-rows" id="<portlet:namespace />_editorNames_field">
+
+	<%
+	for (String editorName : editClientExtensionEntryDisplayContext.getStrings(editorConfigurationCET.getEditorNames())) {
+	%>
+
+		<div class="lfr-form-row">
+			<aui:field-wrapper cssClass="form-group">
+				<aui:input ignoreRequestValue="<%= true %>" label="editor-name" name="editorNames" type="text" value="<%= editorName %>" />
+
+				<div class="form-text form-text-repeat">
+					<liferay-ui:message key="enter-editor-name-to-apply-this-editor-configuration" />
+				</div>
+			</aui:field-wrapper>
+		</div>
+
+	<%
+	}
+	%>
+
+</div>
+
+<div class="lfr-form-rows" id="<portlet:namespace />_editorConfigKeys_field">
+
+	<%
+	for (String editorConfigKey : editClientExtensionEntryDisplayContext.getStrings(editorConfigurationCET.getEditorConfigKeys())) {
+	%>
+
+		<div class="lfr-form-row">
+			<aui:field-wrapper cssClass="form-group">
+				<aui:input ignoreRequestValue="<%= true %>" label="editor-config-key" name="editorConfigKeys" type="text" value="<%= editorConfigKey %>" />
+
+				<div class="form-text form-text-repeat">
+					<liferay-ui:message key="enter-editor-config-key-to-apply-this-editor-configuration" />
+				</div>
+			</aui:field-wrapper>
+		</div>
+
+	<%
+	}
+	%>
+
+</div>
```

Comment:
> If we don't mind adding editor ID to the CX form, then there is a much simpler solution, where we can directly query CX from the context of editor tag or JSP.

Let me clarify what I mean by much simpler solution. I believe we can completely bypass the `ConfigContributor` system. In [EditorTag](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-editor/frontend-editor-taglib/src/main/java/com/liferay/frontend/editor/taglib/servlet/taglib/EditorTag.java#L329) we could have something like
```java
@Override
protected void setAttributes(HttpServletRequest httpServletRequest) {
	setNamespacedAttribute(
		httpServletRequest, "cxURL",
		(CKEditorCET)_cetManager.getCET(...).getURL();

}
```
At that point in code, we know all three variables needed to pinpoint CET entry: editor name, editor config key and portlet id.

Reply:
If I understood, your proposal here is to use this as backend side CX injection point rather than the [CETEditorConfigContributor](https://github.com/liferay-frontend/liferay-portal/pull/3835/commits/5b98a0cfb738cb39fcc2f6b10296a79e8963bc1b#diff-4cb742f7f68026327a08f3ea1fb9243b2acce7213c6c148bda213de79b97fad4R25) right?

Under that assumption, I'd need to understand 2 things:

- Would the declaration of CX be the same (i.e. based on `portlet name`, `editor name` and `editor config key`)?
- How info in the CX set be used to match the current editor being rendered? This is just what [ConfigContributor does](https://github.com/liferay/liferay-portal/blob/b15961e9cfc1b9afdc02d7b09db0f6f2715950da/portal-kernel/src/com/liferay/portal/kernel/editor/configuration/BaseEditorProvider.java#L54-L55) for us, I understand we shall re-implement something similar here.

Reply:
> If I understood, your proposal here is to use this as backend side CX injection point rather than the [CETEditorConfigContributor](https://github.com/liferay-frontend/liferay-portal/pull/3835/commits/5b98a0cfb738cb39fcc2f6b10296a79e8963bc1b#diff-4cb742f7f68026327a08f3ea1fb9243b2acce7213c6c148bda213de79b97fad4R25) right?

Yes.

> Would the declaration of CX be the same (i.e. based on portlet name, editor name and editor config key)?

Not necessarily. The same solution can be applied if the declaration is partially in instance settings, or anywhere else. For this approach to work, the only contract that must be fulfilled is that we can resolve a unique CX definition using locally available `portlet id`, `editor name` and `editor config key`. We can do that in one step, if all three are direct part of CX definition. We can do that in two steps, if we need to first resolve CX ERC from instance settings, and then use that to get URL that is in CX definition.

> How info in the CX set be used to match the current editor being rendered? This is just what [ConfigContributor does](https://github.com/liferay/liferay-portal/blob/b15961e9cfc1b9afdc02d7b09db0f6f2715950da/portal-kernel/src/com/liferay/portal/kernel/editor/configuration/BaseEditorProvider.java#L54-L55) for us, I understand we shall re-implement something similar here.

As far as I'm concerned, if we can have `portlet id`, `editor name` and `editor config key` all in CX definition, great! That results in the the simplest solution. How exactly to match is a matter of backend implementation details. I doubt there a direct API, but we surely can get all CX of the new ckeditor(4) type, iterate through them and find an entry with all three attributes matching.

Reply:
> How exactly to match is a matter of backend implementation details

I see value in providing the same matching logic alrady familiar to implementors of existing third party ConfigContributor (OSGi-based), as it facilitates migration to CX. 

It's true that we're changing things a bit, because now we introduce a new moment where config is added: the frontend. So, things are now a bit different:

- First, all backend configcontributors add their stuff to the config object
- That object is grabbed from the editor jsp implementation
- [new] Frontend logic takes it and adds the CX config contributors
- The end result is sent to the editor instance

Other than that, the thing I like from this approach is that we leverage existing frameworks to build this.

Adding the config in the Editor tag is another possibility, though it gives us some extra work as the matching algorithm also defines an order of priority depending of how many of the 3 things match. That ends up in an config contributor execution order, reflected in the `urls` CX array we send to the front end.

It's debatable whether that level of complexity, related to the order of CX execution for the same target editor, is really needed or not. The point is that this is how backend (osgi-based) config contributors work at this moment. Therefore, if we mirror the behavior, it sounds natural to keep the matching logic already provided.

If we decide to target editors just by Id, then things change and we can just use a matching based on editor id instead. In that case, acting on the EditorTag sounds the right path.

Reply:
> Adding the config in the Editor tag is another possibility, though it gives us some extra work as the matching algorithm also defines an order of priority depending of how many of the 3 things match. That ends up in an config contributor execution order, reflected in the urls CX array we send to the front end.

We don't need to worry about the execution order, because we know CX will resolve last. The fact that we are storing URL dictates this. It must resolve in frontend, therefore it must be last. It would be a different case if we stored config JSON in CX, then priority would be something to consider in backend as well.

Reply:
> We don't need to worry about the execution order, because we know CX will resolve last.

True, they resolve last. 

Nevertheless, more than 1 CX can target the same editor instance. So, amongst all the applicable CX there is also an order. That order is defined by the same priority algorithm as the back-end congributors, as long as we leverage the framework. So, backend execution order of the various [CETConfigContributor](https://github.com/liferay-frontend/liferay-portal/pull/3835/commits/5b98a0cfb738cb39fcc2f6b10296a79e8963bc1b#diff-4cb742f7f68026327a08f3ea1fb9243b2acce7213c6c148bda213de79b97fad4R76-R84) for a target editor will determine [front-end order](https://github.com/liferay-frontend/liferay-portal/pull/3835/commits/5a5a28e91744e866546db790b13d5363cc2248cf#diff-a610b6b30c00c872ffc7a7dc8091bff0e105d37e90e1ab29fe071b7622add7a1R19). This is one of the "exotic" things in this proposal.

Reply:
Fair enough. Priority in overlapping CXs is an issue we never considered before. The same issue will be in all other CXs. For example, there could be multiple FDS Cell Renderer CXs with the same `name`. We will end up applying only one of them at random, at the mercy of backend getter. The same goes for two Java `ConfigContributors` with the same three attributed. I honestly don't think we should cover this case, it's pretty edge. I also don't see how this PR fixes the problem :thinking:

Reply:
Yep, the 3 same attributes is even a more edge case I did not contemplate either. The less edge case is to have, for example, two CX as follows:
* CX1 declares `editor name: ckeditor`
* CX2 declares `editor name: ckeditor, portlet name: com_liferay_..._JournalPortlet`

So, when running the ConfigContributors for a given editor, we'll have 3 cases (i) non-ckeditor instances won't run any, (ii) ckeditor in knowledge base portlet will run CX1 and (iii) ckeditor in web content would run both CX1 and CX2, in that order.

According to the [EditorContributorProvider](https://github.com/liferay/liferay-portal/blob/b15961e9cfc1b9afdc02d7b09db0f6f2715950da/portal-kernel/src/com/liferay/portal/kernel/editor/configuration/BaseEditorProvider.java#L75-L81), it's the serviceRanking which breaks the tie if priority is the same. So, that attribute could be used also in the CX.

Still there may be such a situation (2 CX with the same 3 attrs). As a last resort, yes, that would end up depending on an order determined by the implementation. So this PR is not doing anything new to fix this, it just "mirrors" whatever was happening before, bringing it to the frontend side

I also think we should not worry about that situation